File: electronic_backend/src/main/java/com/project/electronic/Model/Product_Variant.java
package com.project.electronic.Model;

import jakarta.persistence.*;
import jakarta.persistence.criteria.CriteriaBuilder;

@Entity
@Table(name = "product_variant")
public class Product_Variant {

    @Id
    private Integer sku;
    @Column(name = "Name")
    private String Name;
    @Column(name="Price")
    private Integer price;
    @ManyToOne
    @JoinColumn(name = "Product_Id")
    Product product;
    @Column(name = "Quantity")
    private Integer Quantity;

    public Integer getSku() {
        return sku;
    }

    public String getName() {
        return Name;
    }

    public Integer getPrice() {
        return price;
    }

    public Integer getProduct_id() {
        return product.getId();
    }

    public Integer getQuantity() {
        return Quantity;
    }
}

File: Electronic_Frontend/app/src/main/java/com/example/electronic_frontend/model/Product_Variant.kt
package com.example.electronic_frontend.model

class Product_Variant {

    private val sku: Int? = null

    private val Name: String? = null

    private val price: Int? = null

    var product: Product? = null

    private val Quantity: Int? = null

    fun getSku(): Int? {
        return sku
    }

    fun getName(): String? {
        return Name
    }

    fun getPrice(): Int? {
        return price
    }

    fun getProduct_id(): Int? {
        return product!!.getId()
    }

    fun getQuantity(): Int? {
        return Quantity
    }

}